syntax = "proto2";

package uep.protobuf;

message ClientParameters {
    repeated uint32 Ks = 1 [packed=true];
    optional double c = 2;
    optional double delta = 3;
    repeated uint32 RFs = 4 [packed=true];
    optional uint32 EF = 5;
    optional bool ACK = 6;
    optional uint64 fileSize = 7;
    optional bytes header = 8;
    optional uint32 headerSize = 9;
}

enum StartStop {
    START = 0;
    STOP = 1;
}

message ControlMessage {
    oneof carried_message {
	string stream_name = 1;
	ClientParameters client_parameters = 2;
	uint32 client_port = 3;
	uint32 server_port = 4;
	StartStop start_stop = 5;
    }
}
